### ISSUE TITLE
Memory interface support

### ISSUE THREAD

**Issue by arnfol** [esynr3z/corsair#39]
The one feature I am missing the most in corsair is a memory interface. It would be very nice to have the ability to tie address range to a memory while having some of the registers. Such a feature exists in [airhdl](https://airhdl.com/docs/user-guide/concepts/register-map-elements#memories).

**Comment by esynr3z** [comment]
Yeah, that's a nice feature, I've also missed it when used corsair for some tiny lattice project and wanted to access some BRAM directly mapped into interface.
